In [3]:
# copy manually annotated images
import os
import os.path
import shutil
import fnmatch
list_of_dirs_to_copy = ['D:/AI Course/AI Capstone Project/11.AffectNet/Manual/Manually_Annotated/Manually_Annotated_Images'] # List of source dirs
#excluded_subdirs = ['dir1', 'dir2']  # subdir to exclude from copy
dest_dir = 'D:/AI Course/AI Capstone Project/11.AffectNet/Manual/Manually_Annotated/Manually_Annotated_Images/All'     # folder for the destination of the copy
files_patterns = ['*.jpg','*.jpeg','*.png']
for root_path in list_of_dirs_to_copy:
    for root, dirs, files in os.walk(root_path): # recurse walking
        #for dir in excluded_subdirs:
            #if dir in dirs:
                #dirs.remove(dir)   # remove the dir from the subdirs to visit
        if not os.path.exists(dest_dir):
            os.makedirs(dest_dir)  # create the dir if not exists
        for pattern in files_patterns:
            for thefile in fnmatch.filter(files, pattern):  # filter the files to copy
                shutil.copy2(os.path.join(root, thefile), dest_dir) #copy file

SameFileError: 'D:/AI Course/AI Capstone Project/11.AffectNet/Manual/Manually_Annotated/Manually_Annotated_Images\\All\\0000792211b64f6a59bd2d95fee49eabe6373ec1d88f2285bae2ba9b.jpg' and 'D:/AI Course/AI Capstone Project/11.AffectNet/Manual/Manually_Annotated/Manually_Annotated_Images/All\\0000792211b64f6a59bd2d95fee49eabe6373ec1d88f2285bae2ba9b.jpg' are the same file

In [14]:
# copy automatically annotated images
import os
import os.path
import shutil
import fnmatch
list_of_dirs_to_copy = ['D:/AI Course/AI Capstone Project/11.AffectNet/Automatically/Automatically_Annotated(Affect Net)/Automatically_Annotated_Images'] # List of source dirs
#excluded_subdirs = ['dir1', 'dir2']  # subdir to exclude from copy
dest_dir = 'D:/AI Course/AI Capstone Project/11.AffectNet/Automatically/Automatically_Annotated(Affect Net)/Automatically_Annotated_Images/All'     # folder for the destination of the copy
files_patterns = ['*.jpg','*.jpeg','*.png']
for root_path in list_of_dirs_to_copy:
    for root, dirs, files in os.walk(root_path): # recurse walking
        #for dir in excluded_subdirs:
            #if dir in dirs:
                #dirs.remove(dir)   # remove the dir from the subdirs to visit
        if not os.path.exists(dest_dir):
            os.makedirs(dest_dir)  # create the dir if not exists
        for pattern in files_patterns:
            for thefile in fnmatch.filter(files, pattern):  # filter the files to copy
                shutil.copy2(os.path.join(root, thefile), dest_dir) #copy file

SameFileError: 'D:/AI Course/AI Capstone Project/11.AffectNet/Automatically/Automatically_Annotated(Affect Net)/Automatically_Annotated_Images\\All\\0000158c5e783d66117f192303bb9946af28d7a24c5e65a7254c5457.jpg' and 'D:/AI Course/AI Capstone Project/11.AffectNet/Automatically/Automatically_Annotated(Affect Net)/Automatically_Annotated_Images/All\\0000158c5e783d66117f192303bb9946af28d7a24c5e65a7254c5457.jpg' are the same file

In [5]:
#process for Manual Annotated Images
import pandas as pd

m_angry = pd.read_table("m_angry.txt", names=['file'])
m_disgust = pd.read_table("m_disgust.txt", names=['file'])
m_happy = pd.read_table("m_happy.txt", names=['file'])
m_sad = pd.read_table("m_sad.txt", names=['file'])
m_surprised = pd.read_table("m_surprised.txt", names=['file'])
m_neutral = pd.read_table("m_neutral.txt", names=['file'])
m_fear = pd.read_table("m_fear.txt", names=['file'])

#df1_angry = angry.append(angry)
#df2_disgust = disgust.append(disgust)
#df3_happy = happy.append(happy)
#df4_sad = sad.append(sad)
#df5_surprised = surprised.append(surprised)

#data_wext.to_csv("allemotions.csv", index=0)

In [12]:
print(m_fear)

                                                   file
0     00fdd466836fec3e173f5c9d44fdf6305eae913c0be73c...
1     c598d5afe4277c3cbb49bc6801fac0f1ddec2afa6bf6c0...
2     a0c7654bc4bec7a428e32d188ada535cefe846759bbdcc...
3     2a44a45d7474f3df8b40cfe18cc5b9565a119de48f2d02...
4     6024539ed13ae789753b6ac71a3b0dccac533b0664c67b...
...                                                 ...
6373  1239cc905dca360cd611535a1ccd77cd86cc51a5c393bf...
6374  70ba3ba03b1bf3aaee1d834a4b35e17447256670cfd1e4...
6375  2f95cdc69425847851e8fabe8c8a704c02e33afc03cbbf...
6376  a802141c2e6f4861438d6312014b2e55253728905e06ad...
6377  74cff7b8f0d887f0f9822efafb150561169b2a449afe88...

[6378 rows x 1 columns]


Manual Only - In the List vs actual (in folder)
angry: 24,882 (actual: 24,876)
disgust: 3,803 (actual: 3,803)
happy: 134,413 (actual: 134,392)
sad: 25,459 (actual: 25,450
surprised: 14089 (actual: 14,086)
neutral: 74874 (actual: 74,859)
fear: 6378 (actual: 6,375)
Total: 283898 images (actual:  283,841)

In [ ]:
#shutil move (manual files)
import os
import shutil

source = 'D:/AI Course/AI Capstone Project/11.AffectNet/Manual/Manually_Annotated/Manually_Annotated_Images/All'

angry_folder = 'D:/AI Course/AI Capstone Project/11.AffectNet/Manual/angry'
disgust_folder = 'D:/AI Course/AI Capstone Project/11.AffectNet/Manual/disgust'
happy_folder = 'D:/AI Course/AI Capstone Project/11.AffectNet/Manual/happy'
sad_folder = 'D:/AI Course/AI Capstone Project/11.AffectNet/Manual/sad' #train folder
surprised_folder = 'D:/AI Course/AI Capstone Project/11.AffectNet/Manual/surprised'
neutral_folder = 'D:/AI Course/AI Capstone Project/11.AffectNet/Manual/neutral'#train folder
fear_folder = 'D:/AI Course/AI Capstone Project/11.AffectNet/Manual/fear'#train folder

#must change the header "file" specifically, otherwise shutil will not iterate
m_angry_l = m_angry['file'].tolist()
m_disgust_l = m_disgust['file'].tolist()
m_happy_l = m_happy['file'].tolist()
m_sad_l = m_sad['file'].tolist()
m_surprised_l = m_surprised['file'].tolist()
m_neutral_l = m_neutral['file'].tolist()
m_fear_l = m_fear['file'].tolist()

#move to angry
for file in os.listdir(source):
  if file in m_angry_l:
    name = os.path.join(source, file )
    if os.path.isfile(name) :
        shutil.move(name,angry_folder)
    else :
        print ('file does not exist', name)

#move to disgust
for file in os.listdir(source):
  if file in m_disgust_l:
    name = os.path.join(source, file )
    if os.path.isfile(name) :
        shutil.move(name,disgust_folder)
    else :
        print ('file does not exist', name)
        
#move to happy
for file in os.listdir(source):
  if file in m_happy_l:
    name = os.path.join(source, file )
    if os.path.isfile(name) :
        shutil.move(name, happy_folder)
    else :
        print ('file does not exist', name)  
        
#move to sad
for file in os.listdir(source):
  if file in m_sad_l:
    name = os.path.join(source, file )
    if os.path.isfile(name) :
        shutil.move(name, sad_folder)
    else :
        print ('file does not exist', name) 
        
#move to surprised
for file in os.listdir(source):
  if file in m_surprised_l:
    name = os.path.join(source, file )
    if os.path.isfile(name) :
        shutil.move(name, surprised_folder)
    else :
        print ('file does not exist', name)
        
#move to neutral
for file in os.listdir(source):
  if file in m_neutral_l:
    name = os.path.join(source, file )
    if os.path.isfile(name) :
        shutil.move(name, neutral_folder)
    else :
        print ('file does not exist', name)
        
#move to fear
for file in os.listdir(source):
  if file in m_fear_l:
    name = os.path.join(source, file )
    if os.path.isfile(name) :
        shutil.move(name, fear_folder)
    else :
        print ('file does not exist', name)

In [15]:
#process for Automatic Annotated Images
import pandas as pd

a_angry = pd.read_table("a_angry.txt", names=['file'])
a_disgust = pd.read_table("a_disgust.txt", names=['file'])
a_happy = pd.read_table("a_happy.txt", names=['file'])
a_sad = pd.read_table("a_sad.txt", names=['file'])
a_surprised = pd.read_table("a_surprised.txt", names=['file'])
a_neutral = pd.read_table("a_neutral.txt", names=['file'])
a_fear = pd.read_table("a_fear.txt", names=['file'])

#df1_angry = angry.append(angry)
#df2_disgust = disgust.append(disgust)
#df3_happy = happy.append(happy) 
#df4_sad = sad.append(sad)
#df5_surprised = surprised.append(surprised)

#data_wext.to_csv("allemotions.csv", index=0)

In [27]:
print(a_fear)

                                                   file
0     0659295080368f94de5fb33eb8e50276c89d75a8f8bfca...
1     537f30a991271f6542c2f355b6df436571a754c4e73827...
2     1fbaa12a658e14f1a4ed4ac404d506e85de724624f3428...
3     7cc78325b1c1b794d07465540eb781aa21c4514130164e...
4     c5e9e3054f7e97fcde1683f1805a6b340d9b579e4ae7e3...
...                                                 ...
3794  76b0e00de43151a32aa922dd4def271e38405ab5454cd5...
3795  908441ce8b26da1cdbf8c23972a4cf969ab230d8e24680...
3796  bb60181dc3779d6c53d17a630d89346bf2834fb9f8c86f...
3797  07fe4d9ec6d5a750a5e0bcdc4ece03d8620b175e2f7872...
3798  b652c93a3240ed1d69db75ca9cb6ddc35071882f451261...

[3799 rows x 1 columns]


Automatic Only - In the List vs actual (in folder)
angry: 28,000 (actual: 27964)
disgust: 890 (actual: 890)
happy: 246235  (actual: 246078)
sad: 20854 (actual:20838)
surprised: 17462 (actual: 17445)
neutral: 143142 (actual: 143025)
fear: 3799 (actual: 3799)
Total: 460382 images (actual:  460039)

In [17]:
#shutil move (automatically annotated files)
import os
import shutil

source = 'D:/AI Course/AI Capstone Project/11.AffectNet/Automatically/Automatically_Annotated(Affect Net)/Automatically_Annotated_Images/All'

a_angry_folder = 'D:/AI Course/AI Capstone Project/11.AffectNet/Automatically/angry'
a_disgust_folder = 'D:/AI Course/AI Capstone Project/11.AffectNet/Automatically/disgust'
a_happy_folder = 'D:/AI Course/AI Capstone Project/11.AffectNet/Automatically/happy'
a_sad_folder = 'D:/AI Course/AI Capstone Project/11.AffectNet/Automatically/sad' #train folder
a_surprised_folder = 'D:/AI Course/AI Capstone Project/11.AffectNet/Automatically/surprised'
a_neutral_folder = 'D:/AI Course/AI Capstone Project/11.AffectNet/Automatically/neutral'#train folder
a_fear_folder = 'D:/AI Course/AI Capstone Project/11.AffectNet/Automatically/fear'#train folder
 
#must change the header "file" specifically, otherwise shutil will not iterate
a_angry_l = a_angry['file'].tolist()
a_disgust_l = a_disgust['file'].tolist()
a_happy_l = a_happy['file'].tolist()
a_sad_l = a_sad['file'].tolist()
a_surprised_l = a_surprised['file'].tolist()
a_neutral_l = a_neutral['file'].tolist()
a_fear_l = a_fear['file'].tolist()

#move to angry
for file in os.listdir(source):
  if file in a_angry_l:
    name = os.path.join(source, file )
    if os.path.isfile(name) :
        shutil.move(name,a_angry_folder)
    else :
        print ('file does not exist', name)

#move to disgust
for file in os.listdir(source):
  if file in a_disgust_l:
    name = os.path.join(source, file )
    if os.path.isfile(name) :
        shutil.move(name,a_disgust_folder)
    else :
        print ('file does not exist', name)
        
#move to happy
for file in os.listdir(source):
  if file in a_happy_l:
    name = os.path.join(source, file )
    if os.path.isfile(name) :
        shutil.move(name, a_happy_folder)
    else :
        print ('file does not exist', name)  
        
#move to sad
for file in os.listdir(source):
  if file in a_sad_l:
    name = os.path.join(source, file )
    if os.path.isfile(name) :
        shutil.move(name, a_sad_folder)
    else :
        print ('file does not exist', name) 
        
#move to surprised
for file in os.listdir(source):
  if file in a_surprised_l:
    name = os.path.join(source, file )
    if os.path.isfile(name) :
        shutil.move(name, a_surprised_folder)
    else :
        print ('file does not exist', name)
        
#move to neutral
for file in os.listdir(source):
  if file in a_neutral_l:
    name = os.path.join(source, file )
    if os.path.isfile(name) :
        shutil.move(name, a_neutral_folder)
    else :
        print ('file does not exist', name)
        
#move to fear
for file in os.listdir(source):
  if file in a_fear_l:
    name = os.path.join(source, file )
    if os.path.isfile(name) :
        shutil.move(name, a_fear_folder)
    else :
        print ('file does not exist', name)